### ISSUE TITLE
TMP files left after compilations

### ISSUE THREAD

**Issue by GoogleCodeExporter** [tsaiyilien/google-plugin-for-eclipse#74]
```
This is a copy of 
http://code.google.com/p/google-web-toolkit/issues/detail?id=5261

because imho, the solution is in GPE, not gwt compiler


What steps will reproduce the problem?
 Compile GWT projects for hours

What is the expected output? What do you see instead?
 Files left in TMP folder said to cause slowness

My wrk machine is slow anyway & eclipse takes loong time to start.

I made a workaround plugin project to clean the files left in $TMP on start && 
stop of Eclipse.
https://github.com/wadael/GWTCleaner

See comment 26  
http://code.google.com/p/google-web-toolkit/issues/detail?id=5261#c26

I believe you guys can do better than this, as my plugin needs to be started by 
hand (sorry, I'm a plugin newbie). Fortunately, only once. 
Anyway, IMHO this is an easy improvement to do.

Best Regards

Jérôme 
(@wadael)

```

Original issue reported on code.google.com by `wad...@gmail.com` on 3 Jul 2012 at 11:56

**Comment by GoogleCodeExporter** [comment]
```
I just saw this issue; it's an interesting one, as those temporary files are 
created by the GWT compiler (which GPE invokes).

I think the problem is that on Windows, this directory is never cleaned up. I'm 
not sure why it causes Eclipse's startup to slow down though..

We can investigate this one further.
```

Original comment by `rdayal@google.com` on 4 Oct 2012 at 7:01

**Comment by GoogleCodeExporter** [comment]
```
That would be great!
```

Original comment by `tucker...@gmail.com` on 4 Oct 2012 at 7:14

**Comment by GoogleCodeExporter** [comment]
```
Yes, I think only unices clean the TMP.

As I wrote, I am not sure that those files are a cause of slowness for eclipse. 
But I believe the other reporters. 
Anyway, this is at least wasted disk space therefore, my little osgi plugin is 
useful (and now it does not need to be started by hand anymore)

Feel free to add it to your next distribution.
If so tell me so that I celebrate the googlization of my simple code ;)

```

Original comment by `wad...@gmail.com` on 6 Oct 2012 at 7:48

**Comment by GoogleCodeExporter** [comment]
```
Issue 118 has been merged into this issue.
```

Original comment by `danku...@google.com` on 17 Apr 2013 at 7:23

**Comment by GoogleCodeExporter** [comment]
```
Rajeev: the GWT Compiler and DevMode clean after themselves IFF the JVM is 
terminated cleanly. I think this is the main issue: GPE seems to abruptly 
terminate the DevMode rather than sending a message asking it to terminate.
```

Original comment by `t.broyer` on 17 Apr 2013 at 9:24

**Comment by GoogleCodeExporter** [comment]
```
But how about the Terminate and relaunch scenario? I'm (and possibly a lot of 
others) using it when the VM has not enough memory or if dev mode starts to be 
slow (after an hour of debugging/reloading) since its faster and more 
convinient when I'm in the debug perspective. I think Dev mode should keep 
track of all temp files it created and clean them in the next run at least. Or 
there could possibly be some debug hook in GPE that handles the case when the 
Dev mode terminates? Also it happend to me on numerous occasions that the JVM 
crashed resulting in registry amd stack dumps (typicaly some sort of heap or 
stack error probably due to lack of memory).
```

Original comment by `rame...@gmail.com` on 18 Apr 2013 at 6:43

**Comment by GoogleCodeExporter** [comment]
```
Let me repost my comment from Issue 118:

I really really feel that shutdown hook is not enough. There are number of ways 
that hosted application could be terminated, for number of reasons. One reason 
could be fatal error in JVM.

Startup hook should also check for previous startups and clean them up - in 
separate thread, of course.

```

Original comment by `rocky...@gmail.com` on 18 Apr 2013 at 10:12

**Comment by GoogleCodeExporter** [comment]
```
About slow Eclipse startup:
My team members have experienced that behavior when %TEMP% gets cluttered with 
previous GWT startup artifacts. My guess is good as anybody else's but I guess 
that Eclipse for some reason enumerates the contents of %TEMP% for the reasons 
unknown.
```

Original comment by `rocky...@gmail.com` on 18 Apr 2013 at 10:16
